File: web/src/app/models/medicalworker.ts
import {Person} from './person';
import {Speciality} from './speciality';
import {ChatThread} from './chat-thread';

export class MedicalWorker implements Person {

  eMail: string;
  firstName: string;
  id: number;
  preTitle: string;
  lastName: string;
  posTitle: string;
  privateKey: string;
  publicKey: string;
  specialities: Array<Speciality>;
  chatThreads: Array<ChatThread>;
  type: string;
  uid: string;

  constructor() {
    this.specialities = new Array<Speciality>();
    this.chatThreads = new Array<ChatThread>();
  }
}

File: mobile/app/src/main/java/wse18/ase/qse03/mobile/model/MedicalWorker.kt
package wse18.ase.qse03.mobile.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.io.Serializable

@JsonIgnoreProperties(ignoreUnknown = true)
data class MedicalWorker (
    var id: Long,
    var uid: String,
    var preTitle: String?,
    var firstName: String,
    var lastName: String,
    var posTitle: String?,
    var eMail: String?,
    var privateKey: String?,
    var publicKey: String?,
    var speciality: List<Speciality>?,
    var type: MedWorkerType
): Serializable {

    constructor() : this(-1, "","", "", "", "", "", "", "", emptyList<Speciality>(), MedWorkerType.DOCTOR)
}